### ISSUE TITLE
Transition from traditional setup (development setup as outlined in the wiki) to quick-start method docker-compose?

### ISSUE THREAD

**Issue by GlassedSilver** [danbooru/danbooru#4792]
Hello!

So I already have a working instance and would like to switch it from the old method (running "bundle exec rails server --binding 0.0.0.0") to the new docker-compose way.

However, I don't wanna lose my users, posts, wiki and all the other settings like pixiv login credentials and whatever else I configured throughout the time of which I exceedingly have less and less memory because it's been some time.

Is there any graceful way to bring it all over?

This is still a local-only, no FQDN setup for personal use. No proxies, no nothing. :)

Cheers and thank you so much in advance for everyone taking the time to help me!

**Comment by nonamethanks** [comment]
A bit late, but

Mounting here your /var/lib/postgresql/data folder or wherever your installation stores its data in the place of danbooru-data should transfer the SQL data:
https://github.com/danbooru/danbooru/blob/43a665a66d3c5acd7379a593ab51f15986fcbbaa/docker-compose.yaml#L92-L93

As for images, the same should be done for the containers using danbooru-images, which should be replaced with the existing image directory: https://github.com/danbooru/danbooru/blob/43a665a66d3c5acd7379a593ab51f15986fcbbaa/docker-compose.yaml#L71-L73

So for example the latter would become 
```
   -"/path/to/danbooru/public/data:/danbooru/public/data"
```

**Comment by GlassedSilver** [comment]
> A bit late, but
> 
> Mounting here your /var/lib/postgresql/data folder or wherever your installation stores its data in the place of danbooru-data should transfer the SQL data:
> 
> https://github.com/danbooru/danbooru/blob/43a665a66d3c5acd7379a593ab51f15986fcbbaa/docker-compose.yaml#L92-L93
> 
> As for images, the same should be done for the containers using danbooru-images, which should be replaced with the existing image directory:
> 
> https://github.com/danbooru/danbooru/blob/43a665a66d3c5acd7379a593ab51f15986fcbbaa/docker-compose.yaml#L71-L73
> 
> So for example the latter would become
> 
> ```
>    -"/path/to/danbooru/public/data:/danbooru/public/data"
> ```

Those two and my danbooru config file getting mapped should do it I guess. :)

I know where to find the latter, so I'll try switching sometime soon. Maybe when unRAID 6.10 releases, it'd mean I can move my danbooru instance out of my VM into a docker container.

Thank you so much! Don't worry about replying late, it wasn't urgent at all and I'm thankful for the help in any way really. Again thank you a lot!

**Comment by GlassedSilver** [comment]
Hmm, so I did copy the contents of my old setup's PostgreSQL data location (`/var/lib/postgresql/11/main/`) to the docker volume's location.

I was able to test-run the stack using empty volumes, so the images and the environment variables I set and everything works, but I get this in my PostgreSQL's container log repeatedly when trying to deploy the stack with my old data migrated into all relevant locations as you described. 

(and I didn't map my Danbooru config, I just used environment variables in my docker compose. Ah, the beauty of docker-compose. :) )

Bit puzzled now. During my test run with empty volumes however I noticed that deploying with docker-compose already felt that much more controlled and smooth. I'm digging the availability of the docker-compose route a LOT! :)

```
WARNING: POSTGRES_HOST_AUTH_METHOD has been set to "trust". This will allow

         anyone with access to the Postgres port to access your database without

         a password, even if POSTGRES_PASSWORD is set. See PostgreSQL

         documentation about "trust":

         https://www.postgresql.org/docs/current/auth-trust.html

         In Docker's default configuration, this is effectively any other

         container on the same system.

         It is not recommended to use POSTGRES_HOST_AUTH_METHOD=trust. Replace

         it with "-e POSTGRES_PASSWORD=password" instead to set a password in

         "docker run".

********************************************************************************

The files belonging to this database system will be owned by user "postgres".

This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".

The default database encoding has accordingly been set to "UTF8".

The default text search configuration will be set to "english".

Data page checksums are disabled.

initdb: error: directory "/var/lib/postgresql/data" exists but is not empty

If you want to create a new database system, either remove or empty

the directory "/var/lib/postgresql/data" or run initdb

with an argument other than "/var/lib/postgresql/data".
```

**Comment by GlassedSilver** [comment]
Now that I inspected the contents of evazion/postgres, I realized the docker compose stack uses postgresql-14.

My previous setup is from the postgresql-11 days...

Seems like I need to upgrade my postgresql-11 database to 14 schema. Is there a danbooru function I can use for that or would I just use pg_upgrade for that?

Bonus question: Will danbooru's docker compose postgresql container take care of eventual upgrades by itself in the future?
